Controller/OneClick/AbandonedQuote.php

```diff
@@ -0,0 +1,370 @@
+
+    protected $checkoutSession;
+    protected $stateNameMap;
+    protected $cartConverter;
+    protected $cartManagement;
+
+    protected $order;
+    const COD = 'cashondelivery';
+    const RAZORPAY = 'razorpay';
+    const STATE_PENDING_PAYMENT = 'pending_payment';
+
+    /**
+     * CompleteOrder constructor.
+     * @param Http $request
+     * @param Context $context
+     * @param JsonFactory $jsonFactory
+     * @param Data $priceHelper
+     * @param PaymentMethod $paymentMethod
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        Context                                    $context,
+        Http                                       $request,
+        JsonFactory                                $jsonFactory,
+        PaymentMethod                              $paymentMethod,
+        \Razorpay\Magento\Model\Config             $config,
+        \Psr\Log\LoggerInterface                   $logger,
+        \Magento\Quote\Api\CartRepositoryInterface $cartRepositoryInterface,
+        \Magento\Checkout\Model\Session            $checkoutSession,
+        CollectionFactory                          $collectionFactory,
+        StateMap                                   $stateNameMap,
+        CartConverter                              $cartConverter,
+        CartManagementInterface                    $cartManagement,
+        \Magento\Sales\Model\Order                 $order
+    )
+    {
+        parent::__construct($context);
+        $this->request = $request;
+        $this->resultJsonFactory = $jsonFactory;
+        $this->config = $config;
+        $this->rzp = $paymentMethod->setAndGetRzpApiInstance();
+        $this->logger = $logger;
+        $this->cartRepositoryInterface = $cartRepositoryInterface;
+        $this->checkoutSession = $checkoutSession;
+        $this->collectionFactory = $collectionFactory;
+        $this->stateNameMap = $stateNameMap;
+        $this->cartConverter = $cartConverter;
+        $this->cartManagement = $cartManagement;
+        $this->order = $order;
+    }
+
+    public function execute()
+    {
+        $params = $this->request->getParams();
+
+        $resultJson = $this->resultJsonFactory->create();
+
+        $rzpOrderId = $params['rzp_order_id'];
+
+        try {
+            $rzpOrderData = $this->rzp->order->fetch($rzpOrderId);
+
+            $cartMaskId = isset($rzpOrderData->notes) ? $rzpOrderData->notes->cart_mask_id : null;
+
+            $cartId = isset($rzpOrderData->notes) ? $rzpOrderData->notes->cart_id : null;
+            $email = $rzpOrderData->customer_details->email ?? null;
+            $reservedOrderId = isset($rzpOrderData->notes) ? $rzpOrderData->notes->merchant_order_id : null;
+
+            $quote = $this->cartRepositoryInterface->get($cartId);
+
+            $this->updateQuote($quote, $rzpOrderData);
+
+            $quoteId = $rzpOrderData->notes->cart_mask_id;
+
+            $orderPlacement = false;
+
+            try {
```

Comment:
Nested `try ... catch` significantly increases the cyclomatic complexity, making it difficult to troubleshoot / debug